src/clj_github/httpkit_client.clj

```diff
@@ -30,14 +36,16 @@
       (get-in response [:headers "Content-Type"])))
 
 (defn request [client req-map]
-  (let [response @(httpkit/request (prepare client req-map))]
+  (let [request (prepare client req-map)
+        response @(httpkit/request request)]
     (if (success-codes (:status response))
       (update response :body (partial parse-body (content-type response)))
       (throw (ex-info "Request to GitHub failed"
                       {:response (select-keys response [:status :body])}
                       (:error response))))))
 
-(defn new-client [{:keys [app-id private-key token org] :as opts}]
+(defn new-client [{:keys [app-id private-key token org token-fn] :as opts}]
+  {:pre [(or token app-id token-fn)]}
```

Comment:
instead of app-id shouldn't be `(and app-id private-key org)`?

Reply:
Good catch, I think it's actually `(or token token-fn (and app-id private-key))`.